datasketches-memory-java8/src/main/java/org/apache/datasketches/memory/WritableBuffer.java

```diff
@@ -363,10 +366,11 @@ void putBooleanArray(boolean[] srcArray, int srcOffsetBooleans,
 
   //OTHER WRITABLE API METHODS
   /**
-   * For Direct Memory only. Other types of backing resources will return null.
-   * Gets the MemoryRequestServer object used by dynamic off-heap (Direct) memory objects
-   * to request additional memory.
-   * Set using {@link WritableMemory#allocateDirect(long, MemoryRequestServer)}.
+   * For ByteBuffer and Direct Memory backed resources only. Heap and Map backed resources will return null.
+   * Gets the MemoryRequestServer object used by dynamic Memory-backed objects
+   * to request additional memory.  To customize the actions of the MemoryRequestServer,
+   * extend the MemoryRequestServer interfact and
+   * set using {@link WritableMemory#allocateDirect(long, ByteOrder, MemoryRequestServer)}.
```

Comment:
Minor: typo in `interfact`.

Reply:
The original design provided the MemoryRequestServer callback only for Memory segments allocated via `WritableMemory.allocateDirect(...) `calls.  Memory segments allocated via WritableMemory.wrap(ByteBuffer) did not have this capability.  This was a major oversight since all off-heap memory in Druid is allocated using ByteBuffers!  It is unusual that no one has uncovered this until now.  Nonetheless, the fix involves instrumenting all the paths involved in providing this callback mechanism for wrapped ByteBuffers.